pkg/comp-functions/functions/vshnminio/maintenance.go

```diff
@@ -12,25 +14,27 @@ import (
 var service = "minio"
 
 // AddMaintenanceJob will add a job to do the maintenance for the instance
-func AddMaintenanceJob(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+func AddMaintenanceJob(ctx context.Context, svc *runtime.ServiceRuntime) *xfnproto.Result {
 
 	comp := &vshnv1.VSHNMinio{}
-	err := iof.Observed.GetComposite(ctx, comp)
+	err := svc.GetObservedComposite(comp)
 	if err != nil {
-		return runtime.NewFatalErr(ctx, "can't get composite", err)
+		err = fmt.Errorf("cannot get observed composite: %w", err)
+		return runtime.NewFatalResult(err)
 	}
 
-	common.SetRandomSchedules(&comp.Spec.Parameters.Backup, &comp.Spec.Parameters.Maintenance)
+	common.SetRandomSchedules(comp, comp)
```

Comment:
I don't understand this change. The `SetRandomSchedules` function itself didn't change (or has not been committed).
The types of the function argument have changed here, so how can this still work?

Reply:
I've moved the interfaces from the `Parameters.Maintenance` and `Parameters.Backup` to the composition itself.

See: https://github.com/vshn/appcat/pull/103/files/eaf2230e7034f956b0e60a132447088c1662944f#diff-5fb6d956babb9d91192150922edcb8854865f4f84b2b338f744f0ee63483d53cR179

I could however wrap everything in yet another interface to get rid of passing the same object twice.